app/helpers/application_helper.rb

```diff
@@ -226,7 +226,12 @@ def edit_select(object, property, options = {})
 
   def flot_pie_chart(name, title, data, options = {})
     data = data.map { |k,v| {:label=>k.to_s.humanize, :data=>v} } if data.is_a?(Hash)
-    data.map{|element| element[:label] = truncate(element[:label],:length => 16)}
+
+    data.each{|element| element[:label] = truncate(element[:label],:length => 16)}
+
+    # translate data to format expected by statistics charts modules
+    data = data.map{|element| [element[:label], element[:data]]}
+
```

Comment:
you have multiple loops over the same data, while you can join them to be in one loop, i wonder if we should do any truncating server side, or rather let the client display based on available space?